woocommerce/woocommerce
# Stock Filter: Update block to the new structure

> [!WARNING]  
> This PR is based on top of https://github.com/woocommerce/woocommerce/pull/51642


### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:






This PR updates the Stock Filter block to the new structure. Currently, this block supports only the stock status.

Note that the editor logic of the clear button is covered in #51636.

Closes #51638.

### Demo 


https://github.com/user-attachments/assets/4b61eb0a-215c-4294-bca8-6453ebb6a5f2





### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build` command to have experimental blocks available.
2. Add the Product Filters (Experimental) block to the Product Catalog template.
3. Save and go to the shop page.
4. Select some filters, see filter blocks react, and update the filter options accordingly.
5. Interact with the status filter slider, see the filter updated accordingly.
6. Edit the template, and change the status filter block settings.
7. See the editor preview updates.
8. Save the change and see the setting is reflected on the frontend, too.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Experimental: Stock Filter: Update block to the new structure.
</details>
